metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/ObjectsImporter.java

```diff
@@ -144,7 +144,9 @@ public ObjectsImporter(
 
     @Transactional(rollbackFor = Exception.class)
     public Study importStudy(String accession) throws Exception {
+        IMPORT_LOGGER.log(Level.INFO, "Importing study " + accession);
         String xml = sraXmlRetrieverByAccession.getXml(accession);
+        if (xml == null) { return null; }
```

Comment:
I guess it's probably too late to use `Optional` in this project consistently, but in the future consider using it instead of nullable return values.
If you didn't know about [`Optional`](https://mkyong.com/java8/java-8-optional-in-depth/), it forces syntactical checks, improving clarity and reducing the need to remember which values can be null, as it has to be handled explicitly.

Reply:
That's cool, I didn't know about `Optional`! I don't think it's too late to use it consistently. I've included it as part of the ticket mentioned above.